cpp/examples/protobuf/writer.cpp

```diff
@@ -53,13 +53,13 @@ std::string SerializeFdSet(const google::protobuf::Descriptor* toplevelDescripto
   while (!toAdd.empty()) {
     const google::protobuf::FileDescriptor* next = toAdd.front();
     toAdd.pop();
-    next->CopyTo(fdSet.add_file());
+    if (added.find(next->name()) == added.end()) {
+      next->CopyTo(fdSet.add_file());
```

Comment:
Should we have something like `else { continue }` after this, or restructure it as `if (find != end) { continue }`?

Or what about instead of moving the CopyTo inside a conditional, would it work to move `added.insert(next->name())` inside the for loop?